### PR TITLE
NEW: Signal example in utils

### PR DESCRIPTION
**Pull request by pburdet**
Proposition of structure to have internal hyperspy signal examples, for docstr examples and demos prupose

```python

    >>> s = utils.example_signals.EDS_SEM_Spectrum()
    >>> s.plot(True)

```

TO DO
--------

- [x] doc
- [x] tests

figure_eds_sem_spectrum_signal

### CODE REVIEW

File: hyperspy/_signals/eds.py [hunk 1]
```diff
@@ -352,6 +370,28 @@ def add_lines(self,
         only_lines : {None, list of strings}
             If not None, only the given lines will be added.
 
+        Examples
+        --------
+        >>> #add_lines()
```
Comment (to266) [inline hyperspy/_signals/eds.py]: Don't need the line

File: hyperspy/misc/example_signals_loading.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                             'example_signals', '1D_EDS_SEM_Spectrum.hdf5'])
+    return load(file_path)
+
+
+def load_1D_EDS_TEM_spectrum():
+    """
+    Load an EDS-SEM spectrum
+
+    - Sample: FePt bimetallic nanoparticles
+    - SEM Microscope: TecnaiOsiris 200 kV D658 AnalyticalTwin
```
Comment (to266) [inline hyperspy/misc/example_signals_loading.py]: TEM Microscope?

File: hyperspy/utils/__init__.py [hunk 3]
```diff
@@ -13,11 +13,14 @@
         Tools for plotting.
     eds
         Tools for energy-dispersive X-ray data analysis.
+    examples
```
Comment (to266) [inline hyperspy/utils/__init__.py]: `example_signals`

File: doc/user_guide/eds.rst [hunk 4]
```diff
@@ -10,12 +10,32 @@ The methods described here are specific to the following signals:
...
+This chapter described step by step the analysis of an EDS 
 spectrum (SEM or TEM). 
 
+.. NOTE::
+	See also the `EDS tutorials <http://nbviewer.ipython.org/github/hyperspy/hyperspy-	demos/blob/master/electron_microscopy/EDS/>`_ .
+
 Spectrum loading and parameters
 -------------------------------
 
+The files needed for this section can be downloaded using
```
Comment (to266) [inline doc/user_guide/eds.rst]: I would rather say "Data files used in the following examples can be downloaded using". At the moment it looks as if "you HAVE to download these files in order to use these features"

File: doc/user_guide/eds.rst [hunk 5]
```diff
@@ -338,8 +392,21 @@ Selecting certain type of lines:
 Get lines intensity
 -------------------
 
-The :py:meth:`~._signals.eds.EDSSpectrum.get_lines_intensity` 
-method generates intensity maps by peak integration.
+
+The files needed for this section can be downloaded using
```
Comment (to266) [inline doc/user_guide/eds.rst]: Same again

Comment (to266) [inline doc/user_guide/eds.rst]: I would also add a code line in the next example (I assume it's the core-shell sample) in line 415 (where some `spec_img` is used)

File: doc/user_guide/eds.rst [hunk 6]
```diff
@@ -356,14 +423,10 @@ are used by default.
...
-    [<Image, title: Intensity of Al_Ka at 1.49 keV from Spectrum image,
-     dimensions: (|128, 95)>,
-    <Image, title: Intensity of Ni_La at 0.85 keV from Spectrum image,
-     dimensions: (|128, 95)>,
-    <Image, title: Intensity of Ti_Ka at 4.51 keV from Spectrum image,
-     dimensions: (|128, 95)>]
+    >>> s.set_lines(['Fe_Ka', 'Pt_La'])
+    >>> s.get_lines_intensity()
+    [<Image, title: X-ray line intensity of Core shell: Fe_Ka at 6.40 keV, dimensions: (|64, 64)>,
+    <Image, title: X-ray line intensity of Core shell: Pt_La at 9.44 keV, dimensions: (|64, 64)>]
```
Comment (to266) [inline doc/user_guide/eds.rst]: Again, if we are trying to always supply "complete" examples in every snippet, there should be `s` definition here

File: hyperspy/_signals/eds.py (hunk 1 above)
Comment (pburdet) [inline hyperspy/_signals/eds.py]: Done

File: hyperspy/misc/example_signals_loading.py (hunk 2 above)
Comment (pburdet) [inline hyperspy/misc/example_signals_loading.py]: Done

File: hyperspy/utils/__init__.py (hunk 3 above)
Comment (pburdet) [inline hyperspy/utils/__init__.py]: Done

File: doc/user_guide/eds.rst (hunk 4 above)
Comment (pburdet) [inline doc/user_guide/eds.rst]: Done

File: doc/user_guide/eds.rst (hunk 5 above)
Comment (pburdet) [inline doc/user_guide/eds.rst]: Done

File: doc/user_guide/eds.rst (hunk 6 above)
Comment (pburdet) [inline doc/user_guide/eds.rst]: Done here and some other place

**Pull request merged**